ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/verkle/worldview/VerkleWorldState.java

```diff
@@ -206,124 +196,77 @@ protected Hash internalCalculateRootHash(
       final VerkleEntryFactory verkleEntryFactory,
+      final Optional<VerkleWorldStateKeyValueStorage.Updater> maybeStateUpdater,
       final DiffBasedValue<VerkleAccount> accountUpdate) {
-
-    if (!accountUpdate.isUnchanged()) {
-      final VerkleAccount priorAccount = accountUpdate.getPrior();
-      final VerkleAccount updatedAccount = accountUpdate.getUpdated();
-      if (updatedAccount == null) {
-        final Hash addressHash = hashAndSavePreImage(accountKey);
-        verkleEntryFactory
-            .generateKeysForAccount(accountKey)
-            .forEach(
-                bytes -> {
-                  System.out.println(
-                      "remove "
-                          + accountKey
-                          + " "
-                          + bytes
-                          + " "
-                          + accountUpdate.getPrior()
-                          + " "
-                          + accountUpdate.getUpdated());
-                  stateTrie.remove(bytes);
-                });
-        maybeStateUpdater.ifPresent(
-            bonsaiUpdater -> bonsaiUpdater.removeAccountInfoState(addressHash));
-      } else {
-        final Bytes priorValue = priorAccount == null ? null : priorAccount.serializeAccount();
-        final Bytes accountValue = updatedAccount.serializeAccount();
-        if (!accountValue.equals(priorValue)) {
-          verkleEntryFactory
-              .generateKeyValuesForAccount(
-                  accountKey,
-                  updatedAccount.getNonce(),
-                  updatedAccount.getBalance(),
-                  updatedAccount.getCodeHash())
-              .forEach(
-                  (bytes, bytes2) -> {
-                    System.out.println(
-                        "add "
-                            + accountKey
-                            + " "
-                            + bytes
-                            + " "
-                            + bytes2
-                            + " "
-                            + updatedAccount.getBalance());
-                    stateTrie.put(bytes, bytes2);
-                  });
-          maybeStateUpdater.ifPresent(
-              bonsaiUpdater ->
-                  bonsaiUpdater.putAccountInfoState(hashAndSavePreImage(accountKey), accountValue));
-        }
-      }
+    if (accountUpdate.isUnchanged()) {
+      return;
+    }
+    if (accountUpdate.getUpdated() == null) {
+      verkleEntryFactory.generateAccountKeysForRemoval(accountKey);
+      final Hash addressHash = hashAndSavePreImage(accountKey);
+      maybeStateUpdater.ifPresent(
+          bonsaiUpdater -> bonsaiUpdater.removeAccountInfoState(addressHash));
+      return;
+    }
+    final Bytes priorValue =
+        accountUpdate.getPrior() == null ? null : accountUpdate.getPrior().serializeAccount();
+    final Bytes accountValue = accountUpdate.getUpdated().serializeAccount();
+    if (!accountValue.equals(priorValue)) {
```

Comment:
do we still need this check if we have now https://github.com/hyperledger/besu/pull/7594/files#diff-a8c0068954d44abe33b85a9a6c2b70309adb959255e417350abe54c858574b62R208 ?